iotfunctions/anomaly.py

```diff
@@ -204,19 +152,53 @@ def execute(self, df):
                 freqsTS = freqsTS * freqsTSb
                 freqsTS[freqsTS == 0] = 1 / self.windowsize
 
-                # Compute energy = frequency * spectral density over time in decibel - no log10
-                ETS = np.dot(SxTS.T, freqsTS)
+                highfreqsTS = freqsTS.copy()
+                lowfreqsTS = freqsTS.copy()
+                highfreqsTS[highfreqsTS <= 0.3] = 0
+                lowfreqsTS[lowfreqsTS > 0.3] = 0
 
-                # compute zscore over the energy
-                ets_zscore = (ETS - ETS.mean())/ETS.std(ddof=0)
-                logger.debug('NoData z-score max: ' + str(ets_zscore.max()))
+                # Compute energy = frequency * spectral density over time in decibel
+                try:
+                    lowETS = np.log10(np.dot(SxTS.T, lowfreqsTS))
+                    highETS = np.log10(np.dot(SxTS.T, highfreqsTS))
+                    ETS = np.log10(np.dot(SxTS.T, freqsTS))
+
+                    # compute the elliptic envelope to exploit Minimum Covariance Determinant estimates
+                    #    standardizing
+                    lowETS = (lowETS - lowETS.mean())/lowETS.std(ddof=0)
+                    highETS = (highETS - highETS.mean())/highETS.std(ddof=0)
+
+                    twoDimETS = np.vstack((lowETS, highETS)).T
+                    logger.info('lowETS: ' + str(lowETS) + ', highETS:' + str(highETS) + 'input' + str(twoDimETS))
+
+                    # inliers have a score of 1, outliers -1, and 0 indicates an issue with the data
+                    dfe[self.output_item] = 0.0002
+                    ellEnv = EllipticEnvelope(random_state=0)
+
+                    dfe[self.output_item] = 0.0003
+                    ellEnv.fit(twoDimETS)
+                    #ets_zscore = ellEnv.predict(twoDimETS)
+                    dfe[self.output_item] = 0.0004
+                    ets_zscore = ellEnv.decision_function(twoDimETS, raw_values=True).copy()
+
+                    # compute zscore over the energy
+                    #ets_zscore = np.abs((ETS - ETS.mean())/ETS.std(ddof=0))
+                    logger.info('Spectral z-score max: ' + str(ets_zscore.max()))
+
+                    # length of timesTS, ETS and ets_zscore is smaller than half the original
+                    #   extend it to cover the full original length 
+                    dfe[self.output_item] = 0.0005
+                    Linear = sp.interpolate.interp1d(timesTS, ets_zscore, kind='linear', fill_value='extrapolate')
```

Comment:
Good to follow variable naming convention in python. example: variable_test_value, varable_example. Only constant will be capital letters.

Reply:
changed the name to linear_interpolate